Update lifecycle event and callback documentation

FoxxMD opened the pull request:
* Clarify the difference between events and callbacks
* Add the listener() method for builder added by laravel-doctrine/fluent#45

FoxxMD committed:
Update lifecycle event and callback documentation

* Clarify the difference between events and callbacks
* Add the listener() method for builder added by laravel-doctrine/fluent#45

FoxxMD commented:
@patrickbrouwers I can make PRs for the other branches if you want. Just let me know!

guiwoda commented on fluent/reference.md:
```diff
@@ -618,14 +618,57 @@ $builder->joinedTableInheritance()->column('type');
-Use `$builder->events()` to access the `LaravelDoctrine\Fluent\Builders\LifecycleEvents` object.
+There are two sets of events that can be used by two different types of listeners:
+
+**Lifecycle Event** Listeners and Subscribers are classes with specific callback methods that receives some kind of EventArgs instance.
```
@FoxxMD can we look for an alternative to the "some kind of" phrase?

> Both Listener and Subscriber classes can hook into **Lifecycle Events**. The event dispatcher will then call on the specific callback method with a specific instance of `EventArgs`, depending on the event type.

guiwoda commented on fluent/reference.md:
```diff
@@ -618,14 +618,57 @@ $builder->joinedTableInheritance()->column('type');
+
+**Lifecycle Event** Listeners and Subscribers are classes with specific callback methods that receives some kind of EventArgs instance.
+
+**Lifecycle Callbacks** are methods on the entity classes that are called when the event is triggered.
```
@FoxxMD if you change phrasing as suggested above, maybe this one could change as well:

> Entities themselves can listen to **Lifecycle Callbacks** and have a method called on them when a specific event is triggered.

FoxxMD committed:
More clarification for lifecycle event and callback documentation

* Remove summary at beginning of section (better explained in each section)
* Change wordage for first section to "listening to..." rather than "Lifecycle Events" to better reflect behavior
* Suggested additions for EventArgs from @guiwoda
* Removed subscriber wordage since fluent does not support subscribers (should add via doctrine config)

FoxxMD commented:
@guiwoda I added more info on EventArgs and also simplified the section. I made a clarification on the first section as really it's about "listeners for Lifecycle Events" rather than (erroneously) "Lifecycle Event triggered by Lifecycle Event" which is kind of how it read previously. Let me know what you think.

patrickbrouwers commented on fluent/reference.md:
````diff
@@ -618,14 +618,55 @@ $builder->joinedTableInheritance()->column('type');
 ```
 
````
Perhaps good to update the index at the top of the page too. Not still refers to the callbacks only.

FoxxMD replied:
This occurred to me too but I wanted to leave intact in case someone has this exact URL bookmarked. Very small chance I guess :)

guiwoda commented on fluent/reference.md:
```diff
@@ -618,14 +618,55 @@ $builder->joinedTableInheritance()->column('type');
+
+### Lifecycle Callback
+
+_Lifecycle Callbacks_ can be triggered for a **subset** of _Lifecycle Events_. A callback is defined on an Entity class and are triggered when an instance of that Entity experiences a relevant Event.
```
(...) and _is_ triggered when an (...)

FoxxMD replied:
fixed, good find.

guiwoda reviewed (changes requested):
A lot better! Just a small typo I found

FoxxMD committed:
Minor lifecycle event and callback documentation cleanup

* Anchor link reflects section name
* Fix typo

FoxxMD reviewed (commented):
Minor changes

FoxxMD commented:
I don't totally understand how reviews were but I pushed both changes 👍
